mat_dp_pipeline/sdf/standard_data_format.py

```diff
@@ -88,6 +91,21 @@ def read(self, path: FileOrPath) -> pd.DataFrame:
         )
 
 
+class SDFMetadata(pydantic.BaseModel):
+    """Metadata of the Standard Data Format.
+
+    Args:
+        main_label (str): Description of what the path in the SDF represents. Defaults to "Location"
+        tail_labels (list[str]):
+            Names of the lowest levels in the SDF. If none are provided, the whole
+            path is described by `main_label`. If provided, for example -- ["Scenario", "Parameter"],
+            it means that the lowest levels in the SDF refer to a name of a Scenario and a Parameter.
+    """
+
+    main_label: str = "Location"
```

Comment:
Is there some way we could perhaps embed it in "SourceWithCountries"?

Reply:
Ok this seems feasible - currently tail headers is taken from the source in the single target case - we can just take main header also from target source in this case. In the case where it can't be identified, perhaps we should default to something more generic like "Select". We add "Country" to SourceWithCountries. Then in the yellow file case, we override the data source to say "Location".

Reply:
https://github.com/dreamingspires/mat-dp-pipeline/pull/12/files